pkg/provider/kubernetes/gateway/kubernetes.go

```diff
@@ -894,7 +1176,7 @@ func loadServices(client Client, namespace string, targets []v1alpha1.HTTPRouteF
 		var portStr string
 		for _, subset := range endpoints.Subsets {
 			for _, p := range subset.Ports {
-				if portName == p.Name {
+				if portSpec.Name == p.Name {
```

Comment:
that's a good cleanup. we should check if we can do the same in other places of the codebase.